instrumentation/servlet/servlet-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/ServletAccessor.java

```diff
@@ -55,6 +59,17 @@ public interface ServletAccessor<REQUEST, RESPONSE> {
 
   List<String> getRequestParameterValues(REQUEST request, String name);
 
+  default void forAllQueryParams(
+      REQUEST request, BiConsumer<AttributeKey<List<String>>, List<String>> consumer) {
+    Map<String, String[]> parameterMap = getQueryParamsMap(request);
+    for (Map.Entry<String, String[]> entry : parameterMap.entrySet()) {
+      AttributeKey<List<String>> key = AttributeKey.stringArrayKey(entry.getKey());
```

Comment:
can these AttributeKeys be cached? e.g. check out `ServletRequestParametersExtractor.parameterAttributeKey()`

Reply:
I did notice that they were cached in the other path, and I do grok that it amounts to allocations on a hot path, but I wasn't sure how important we thought it would be. Seemed like premature optimization for a setting that might not be used that often? 

I dunno, I can add it if we think it's important.

Reply:
I think probably worth caching, given our strong javadoc statement about pre-allocating keys

https://github.com/open-telemetry/opentelemetry-java/blob/bb55bcc736269573b32f5207366ea33af6e4fc15/api/all/src/main/java/io/opentelemetry/api/trace/Span.java#L96-L97